src/python/datatype.rs

```diff
@@ -124,6 +124,22 @@ impl PyDataType {
         .into())
     }
 
+    #[staticmethod]
+    pub fn r#struct(fields: &PyDict) -> PyResult<Self> {
```

Comment:
haven't seen this before what does the `r#` do?

Reply:
This is a "raw identifier", which allows us to define a `struct` method, which would otherwise conflict with the reserved `struct` Rust keyword!

https://doc.rust-lang.org/rust-by-example/compatibility/raw_identifiers.html